lib/routes/account.js

```diff
@@ -972,27 +1062,42 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
-              mailer,
-              sessionToken,
-              sessionToken.emailCode,
-              {
-                service: service,
-                redirectTo: request.payload.redirectTo,
-                resume: request.payload.resume,
-                acceptLanguage: request.app.acceptLanguage
+
+        return db.sessionTokenWithVerificationStatus(sessionToken.id)
+          .then(function (token) {
+
+            // Choose which type of email to send
+            var doSigninConfirmation = requestHelper.shouldEnableSigninConfirmation(sessionToken.uid.toString('hex'), config)
+            var func
+            if (!token.emailVerified || !doSigninConfirmation) {
+
+              // Check for blackout period
+              if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+                return reply({})
               }
-            )
-          )
+
+              func = mailer.sendVerifyCode
+            } else {
+              func = mailer.sendVerifyLoginEmail
+            }
+
+            return customs.check(
+              request.app.clientAddress,
+              sessionToken.email,
+              'recoveryEmailResendCode')
+              .then(func.bind(
+                mailer,
+                sessionToken,
+                doSigninConfirmation ? token.tokenVerificationId : sessionToken.emailCode, // Send tokenVerficationId, unless sign-in confirmation is disabled
```

Comment:
There is a subtle problem here, as noted in the [Sign up using old auth server, with new auth server, click "resend" from "/confirm" screen test notes](https://public.etherpad-mozilla.org/p/fxa-83-confirm-login). If the user signs up with the old auth server, then clicks "resend" from the confirm screen under the new auth server, `sendVerifyCode` is called with a code of `null`. I think this line should be something along the lines of:

```js
doSigninConfirmation ? token.emailVerified ? token.tokenVerificationId: sessionToken.emailCode : sessionToken.emailCode
```

Perhaps that'd be easier to read if the code was set up where `func` is set.